components/nexus-mime/src/test/java/org/sonatype/nexus/mime/NexusMimeTypesTest.java

```diff
@@ -44,26 +42,24 @@ private Properties addMimeType(final Properties properties, final String extensi
 
   @Test
   public void unconfigured() {
-    assertThat(underTest.getMimeTypes("test"), is(nullValue()));
+    assertThat(underTest.getMimeRuleForExtension("test"), is(nullValue()));
   }
 
   @Test
   public void addMimeType() {
     underTest.initMimeTypes(addMimeType(new Properties(), "test", "application/octet-stream"));
-    assertThat(underTest.getMimeTypes("test"), is(notNullValue()));
-    assertThat(underTest.getMimeTypes("test"), hasProperty("extension", is("test")));
-    assertThat(underTest.getMimeTypes("test"), hasProperty("override", is(false)));
-    assertThat(underTest.getMimeTypes("test").getMimetypes(), contains("application/octet-stream"));
+    assertThat(underTest.getMimeRuleForExtension("test"), is(notNullValue()));
```

Comment:
only one call to getMimeRuleForExtension() is needed here, can cache the result for inspection